src/guiguts/tools/jeebies.py

```diff
@@ -0,0 +1,297 @@
+
+            rlim = min(rlim, len(para))
+
+        elif indx == len(para) - 1:
+            # Required slice is the tail of the paragraph string.
+            rlim = len(para)
+            llim = rlim - 60
+
+            llim = max(llim, 0)
+
+        else:
+            # Required slice is to be centered around indx if possible.
+            llim = indx - 30
+            rlim = indx + 30
+
+            if llim < 0:
+                llim = 0
+                rlim = 60
+
+            if rlim > len(para):
+                llim = len(para) - 60
+                rlim = len(para)
+
+            llim = max(llim, 0)
+
+        # As a result of the abitrary width limit chosen above, the start
+        # and end of the slice of the paragrpah string defined by llim and
+        # rlim may have partial words. Expand the slice at both ends until first
+        # whitespace character found or start or end of paragraph encountered.
+        # The slice that is defined will now have only whole words.
+
+        while llim > 0 and para[llim : llim + 1] != " ":
+            llim -= 1
+
+        # If not at left margin set llim to start of first word in slice.
+        if llim != 0:
+            llim += 1
+
+        while rlim < len(para) and para[rlim : rlim + 1] != " ":
+            rlim += 1
+
+        # If not at right margin set rlim to end of last word in slice.
+        if rlim != len(para) - 1:
+            rlim -= 1
+
+        return llim, rlim + 1, para[llim : rlim + 1]
+
+    def find_in_dictionary(self, three_form: str) -> int:
+        """returns frequency in hebe phrase corpus or zero"""
+
+        words = three_form.split()
+        key = words[0] + "|" + words[1] + "|" + words[2]
+        freq = 0
+        if key in self.dictionary:
+            freq = self.dictionary[key]
+
+        return freq
+
+    def add_to_dialog(
+        self, header: str, para_slice: str, checker_dialog: CheckerDialog
+    ) -> None:
+        """helper function that abstracts repeated code
+
+        Args:
+            header: a 3-form phrase with explanatory info
+            para_slice: a slice of a paragraph centered around the 3-form phrase
+            checker_dialog: where report text is written
+        """
+
+        record = "    " + para_slice
+        # Add header and record to the dialog.
+        checker_dialog.add_entry(header)
+        checker_dialog.add_entry(record)
+        checker_dialog.add_entry(" ")
+
+    def load_phrases_file_into_dictionary(self) -> None:
+        """Load a phrase-per-line file into the target dictionary. Entries are
+        either 2-form (|he|a:168) or 3-form (since|he|stopped:7) phrases.
+        """
+        file = "hebelist.txt"
+        path = DEFAULT_DICTIONARY_DIR.joinpath(file)
+        try:
+            with path.open("r", encoding="utf-8") as fp:
+                for line in fp:
+                    line = line.strip()
+                    if ":" in line:
+                        parts = line.split(":")
```

Comment:
You can also do:
```python
key, value = line.split(":")
```

That makes the variables a bit clearer when you use them later.